apps/strawberry_fields_listener.py

```diff
@@ -35,16 +35,15 @@ def __init__(self):
         self.result = []
 
     def run(self):
-        if 'num_subsystems' not in self.device['kwargs']:
-            raise ValueError("Must specify the number of subsystems")
+        if self.target is None:
+            raise ValueError("Blackbird program has no target backend")
 
         self.eng, self.q = sf.Engine(
-            self.device['kwargs']['num_subsystems'],
-            hbar=self.device['kwargs'].get('hbar', 2)
+            len(self.active_modes),
```

Comment:
Is this stuff an unrelated change that snuck into the same PR?

Reply:
Slightly related. Since we no longer store `eng.run()` compatible arguments in the blackbird file, I had to modify this to 'extract' the num_subsystems from the blackbird script, to keep it working.

Reply:
Looks better the new way